public/app/features/dashboard-scene/settings/GeneralSettings.tsx

```diff
@@ -22,11 +69,190 @@ export class GeneralSettingsEditView
+                  <GenAIDashTitleButton onGenerate={onTitleChange} dashboard={dashboard} />
+                )} */}
+                </HorizontalGroup>
+              }
+            >
+              <Input
+                id="title-input"
+                name="title"
+                defaultValue={title}
+                onBlur={(e: ChangeEvent<HTMLInputElement>) => onTitleChange(e.target.value)}
+              />
+            </Field>
+            <Field
+              label={
+                <HorizontalGroup justify="space-between">
+                  <Label htmlFor="description-input">
+                    {t('dashboard-settings.general.description-label', 'Description')}
+                  </Label>
+
+                  {/* {config.featureToggles.dashgpt && (
+                  <GenAIDashDescriptionButton onGenerate={onDescriptionChange} dashboard={dashboard} />
+                )} */}
+                </HorizontalGroup>
+              }
+            >
+              <TextArea
+                id="description-input"
+                name="description"
+                defaultValue={description}
+                onBlur={(e: ChangeEvent<HTMLTextAreaElement>) => onDescriptionChange(e.target.value)}
+              />
+            </Field>
+            <Field label={t('dashboard-settings.general.tags-label', 'Tags')}>
+              <TagsInput id="tags-input" tags={tags} onChange={onTagsChange} width={40} />
+            </Field>
+            <Field label={t('dashboard-settings.general.folder-label', 'Folder')}>
+              <FolderPicker
+                value={meta.folderUid}
+                onChange={onFolderChange}
+                // TODO deprecated props that can be removed once NestedFolderPicker is enabled by default
+                initialTitle={meta.folderTitle}
+                inputId="dashboard-folder-input"
+                enableCreateNew
+                skipInitialLoad
+              />
+            </Field>
+
+            <Field
+              label={t('dashboard-settings.general.editable-label', 'Editable')}
+              description={t(
+                'dashboard-settings.general.editable-description',
+                'Set to read-only to disable all editing. Reload the dashboard for changes to take effect'
+              )}
+            >
+              <RadioButtonGroup value={editable} options={EDITABLE_OPTIONS} onChange={onEditableChange} />
+            </Field>
+          </Box>
+
+          <TimePickerSettings
+            onTimeZoneChange={onTimeZoneChange}
+            onWeekStartChange={onWeekStartChange}
+            onRefreshIntervalChange={onRefreshIntervalChange}
+            onNowDelayChange={onNowDelayChange}
+            onHideTimePickerChange={onHideTimePickerChange}
+            onLiveNowChange={onLiveNowChange}
+            refreshIntervals={refreshPicker?.state?.intervals}
+            // TODO: Control visibility of time picker
+            // timePickerHidden={timepicker?.state?.hidden}
```

Comment:
Not trivial. Since the visibility now is controlled by adding the scene object or not when the scene is created. If we remove the node, we cannot recover the previous state. So, once is hidden and re-enabled, the previous state of the object is lost.

Reply:
Instead of doing this in for DashboardControls:

```
timeControls: Boolean(oldModel.timepicker.hidden)
  ? []
  : [
    new SceneTimePicker({}),
    new SceneRefreshPicker({
      refresh: oldModel.refresh,
      intervals: oldModel.timepicker.refresh_intervals,
    }),
  ],
```

we could introduce `DashboardControls.state.isTimeControlsHidden` and do:

```
isTimeControlsHidden: Boolean(oldModel.timepicker.hidden),
timeControls: [
    new SceneTimePicker({}),
    new SceneRefreshPicker({
      refresh: oldModel.refresh,
      intervals: oldModel.timepicker.refresh_intervals,
    }),
  ],
```

and render time controls accordingly to the state. This way we can easily preserve state of the corresponing objects while still making them hidden if necessary.